iceoryx_utils/include/iceoryx_utils/internal/relocatable_pointer/relative_ptr.hpp

```diff
@@ -234,13 +234,13 @@ class relative_ptr : public RelativePointer
 
 
     relative_ptr(const RelativePointer& other)
+        : RelativePointer(other)
     {
-        m_offset = computeOffset(other.computeRawPtr());
     }
 
     relative_ptr& operator=(const RelativePointer& other)
     {
-        m_offset = computeOffset(other.computeRawPtr());
+        *this = other;
```

Comment:
Where is the test that reveals that this is an infinite recursion now? @marthtz

Reply:
yeah, it seems we are missing some tests ... `*static_cast<RelativePointer*>(this) = other;` should do the trick ... I guess I'm too dependent on my IDE to prevent me from doing stupid things :D

Reply:
Added a test and fixed it.
However, I'm using `RelativePointer::operator=(other);` as this is IMHO straightforward  and easier to comprehend than `static_cast<...`.